### PR TITLE
stop relying on giant dictionary for help parsing

### PR DESCRIPTION
**Pull request by awsomearvinder**

### CODE REVIEW

File: src/cogs/help.py [hunk 1]
```diff
@@ -160,57 +93,55 @@ async def send_help(self, ctx, *args):
                 color=self.embed_color,
             )
 
-            for cog in cog_dict:
+            for cog in self.bot.cogs:
                 with suppress(Exception):  # forced to use this because pylint
                     embed.add_field(
                         name=cog,
-                        value=f'`{cog_dict[cog]["command_count"]} commands`',
+                        value="`N commands`",
```
Comment (Ganoosh) [inline src/cogs/help.py:100]: What is `N` here?

File: src/cogs/help.py [hunk 2]
```diff
@@ -9,107 +9,48 @@
...
             alias_string += f"`{alias}`, " if alias else None
         alias_string = alias_string[:-2]
         dash = ", " if alias_string else ""
-        usage = doc_item["usage"] if doc_item["usage"] else None
+        usage = cmd.usage if cmd.usage else ""
         description_string += (
-            f"`{item}`{dash}{alias_string}:\n"
-            f'{doc_item["doc_string"]}\n'
-            f"```{prefix}{item} {usage}```\n\n"
+            f"`{cmd}`{dash}{alias_string}:\n"
```
Comment (Ganoosh) [inline src/cogs/help.py:46]: Wouldn't `cmd` in this case return an object? `cmd.name` would fit better, wouldn't it?

File: src/cogs/help.py (hunk 1 above)
Comment (awsomearvinder) [reply src/cogs/help.py:100]: Completely forgot to get rid of that! I just hadn't figured out how to count the enabled commands, I'll add that in tomorrow.

File: src/cogs/help.py [hunk 3]
```diff
@@ -9,107 +9,48 @@
...
         dash = ", " if alias_string else ""
-        usage = doc_item["usage"] if doc_item["usage"] else None
+        usage = cmd.usage if cmd.usage else None
         description_string += (
-            f"`{item}`{dash}{alias_string}:\n"
-            f'{doc_item["doc_string"]}\n'
-            f"```{prefix}{item} {usage}```\n\n"
+            f"`{cmd.name}`{dash}{alias_string}:\n"
+            f"{doc_item}\n"
+            f"```{prefix}{cmd} {usage}```\n\n"
```
Comment (Ganoosh) [inline src/cogs/help.py:48]: Why is the top mention `cmd.name` but the bottom one just `cmd`?

**Pull request merged**